### PR TITLE
Spreadsheet Automation

### LINKED ISSUE #469: Spreadsheet Automation
# Aim

The python script will automate the spreadsheet

# Details

This will compare the data of two spreadsheets and give a pie chart.

### PR DESCRIPTION
**Pull request by Amit366**
# Description

The script will compare two spreadsheets and then show a pie chart

### CODE REVIEW

**Comment by Amit366** [comment]
@avinashkranjan please add the level tag

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
@Amit366 Please add a README.md with instruction on how to run the script

**Review by antrikshmisri — APPROVED** [review]
Approved from my side. @kaustubhgupta please add the tags

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
This is not a generalized script. What happens if I select totally different datasets that do not have "Total Price" and "Purchased Amount" columns? Here is the solution to this:

- Make it a CLI version where the user mentions the dataset 1 column to be compared and the same for dataset 2. 
- Take the input for the common column where the merger will happen.
- Now display the plots

**Comment by Amit366** [comment]
@kaustubhgupta sir I have updated it

File: Spreadsheet_Automation/script.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# reading the data
+data_prices = pd.read_excel(book_prices)
+data_home_1 = pd.read_excel(book_relative_path)
+
+#print​(df_prices, df_home_1)
+
+item = input("What is the basis of merging? ")
+data_total = data_home_1.merge(data_prices, on=item)
+
+data_total['Total Price'] = data_total['PURCHASED AMOUNT'] * data_total['Price']
```
Comment (kaustubhgupta) [inline Spreadsheet_Automation/script.py:19]: This line is not required I think?

File: Spreadsheet_Automation/script.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+#print​(df_total)
+material=input("Enter criteria 1")
```
Comment (kaustubhgupta) [inline Spreadsheet_Automation/script.py:22]: name the variables with descriptive meaning. Suggestion:
```suggestion
criteria_1=input("Enter criteria 1")
```

File: Spreadsheet_Automation/script.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+price=input("Enter criteria 2")
+fig = px.pie(data_total[[material, price]], values=price, names=material)
```
Comment (kaustubhgupta) [inline Spreadsheet_Automation/script.py:24]: Update the changes here also

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by Amit366** [comment]
@kaustubhgupta now check

File: Spreadsheet_Automation/script.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# importing libraries
+
+import pandas as pd
+import plotly.express as px
+
+# storing the dataset
+book_relative_path = input("Enter first dataset")
```
Comment (kaustubhgupta) [inline Spreadsheet_Automation/script.py:7]: These two datasets names also need to be changed to something good.

File: Spreadsheet_Automation/script.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+book_prices = input("Enter second dataset")
+
+# reading the data
+data_prices = pd.read_excel(book_prices)
```
Comment (kaustubhgupta) [inline Spreadsheet_Automation/script.py:11]: Here also

File: Spreadsheet_Automation/script.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+data_home_1 = pd.read_excel(book_relative_path)
+
+#print​(df_prices, df_home_1)
+
+item = input("What is the basis of merging? ")
```
Comment (kaustubhgupta) [inline Spreadsheet_Automation/script.py:16]: And this item too.

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Sorry I didn't notice these variables earlier:

**Comment by Amit366** [comment]
@avinashkranjan updated

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@santushtisharma10 your approval awaited

**Review by santushtisharma10 — APPROVED** [review]

**Comment by avinashkranjan** [comment]
@all-contributors please add @Amit366  for code and documentation

**Pull request merged**